### ISSUE TITLE
LG-282: MicroProfile Health 3.0 Implementation (part of MP 4.0)

### ISSUE THREAD

**Issue by donbourne** [OpenLiberty/open-liberty#11472]
Describe the high level feature, including any external spec links.

When ready, add links to the Upcoming Feature Overview document and Feature Test Summary issue:
- UFO: https://ibm.box.com/s/fqgihmbj5oo1sfoqirjplbsgnfo7yjrj
- FTS: https://github.com/OpenLiberty/open-liberty/issues/14981
- OL ID: https://github.com/OpenLiberty/docs/issues/3292

- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
## List of Steps to complete or get approvals / sign-offs for Onboarding to the Liberty release (GM date)

Instructions:
- Do the actions below and mark them complete in the checklist when they are done.
- Make sure all feature readiness approvers put the appropriate tag on the epic to indicate their approval.

## **Design**
Before Development Starts or 8 weeks before Onboarding
- [x] POC Design / UFO Review Scheduled (David Chang) or N/A.
- [x] POC Design / UFO Reviewed (Feature Owner) or N/A.
- [x] Complete any follow-ons from the POC Review.
- [x] Design / UFO Approval (Alasdair Nottingham) or N/A.
- [x] No Design / No UFO Approval (Arthur De Magalhaes - cloud / Alasdair Nottingham - server) or N/A.
- [x] SVT Requirements identified. (Epic owner / Feature owner with SVT focal point)
- [x] ID Requirements identified. (Epic owner / Feature owner with ID focal point)
- [x] Create a child task of this epic entitled "FAT Approval Test Summary". Add the link in above.

## **Legal**
3 weeks before Onboarding
- [x] Identify all open source libraries that are changing or are new. Work with Legal Release Services (Cass Tucker or Release PM) to get open source cleared and approved. Or N/A. (Epic Owner).   New or changed open source impacts license and Certificate of Originality.

## **Translation**
3 weeks before Onboarding
- [x] All new or changed PII messages are checked into the integration branch, before the last translation shipment out. (Epic Owner)

## **Feature Complete**
2 weeks before Onboarding
- [x] Implementation complete. (Epic owner / Feature owner)
- [x] All function tests complete. Ready for FAT Approval. (Epic owner / Feature owner)
- [x] Review all known issues for Stop Ship. (Epic owner / Feature owner / PM)

## **Focal Point Approvals**
2 to 1 week before Onboarding
#### You **MUST** have the Design Approved or No Design Approved label before requesting focal point approvals.

All features (both "Design Approved" and "No Design Approved")
- [x] **FAT** - (Kevin Smith). SOE FATS are running successfully or N/A . Approver adds label focalApproved:fat to the Epic in Github.
- [x] **Demo** - (Tom Evans or Chuck Bridgham). Demo is scheduled for an upcoming EOI. Approver adds label focalApproved:demo to the Epic in Github.
- [x] **Globalization** (Sam Wong - Liberty / Simy Cheeran - tWAS). Translation is complete or N/A. TVT - complete or N/A. Approver adds label focalApproved:globalization to the Epic in Github.

"Design Approved" features
- [x] **Accessibility** - (G Scott Johnston). Accessibility testing is complete or N/A. Approver adds label focalApproved:accessibility to the Epic in Github.
- [x] **ID** - (Kareen Deen). Documentation work is complete or N/A . Approver adds label focalApproved:id to the Epic in Github.
- [x] **Performance** - (Jared Anderson). Performance testing is complete with no high severity defects or N/A . Approver adds label focalApproved:performance to the Epic in Github.
- [x] **Serviceability** - (Don Bourne). Serviceability has been addressed.
- [x] **STE** - (Swati Kasundra). STE chart deck is complete or N/A . Approver adds label focalApproved:ste to the Epic in Github.
- [x] **SVT** - (Greg Ecock - Cloud, Brian Hanczaryk- APS). SVT is complete or N/A . Approver adds label focalApproved:svt to the Epic in Github.

## **Ready for GA**
1 week before Onboarding
- [x] No Stop Ship issues for the feature. (Epic owner / Feature owner / Release PM)
- [x] Ship Readiness Review and Release Notes completed (Epic owner / Feature owner / Release PM)
- [ ] Github Epic and Epic's issues are closed / complete. All PRs are committed to the master branch. (Epic owner / Feature owner / Backlog Subtribe PM)

## **Other deliverbles**
- [x] **OL Guides** - (Yee-Kang Chang). Assessment for OL Guides is complete or N/A.
- [x] **WDT** - (Leonard Theivendra). WDT work complete or N/A.
- [x] **Blog** - (Laura Cowen) Blog article writeup (Epic owner / Feature owner / Laura Cowen)

**Comment by pgunapal** [comment]
@steven1046 Hi Steven, There are no addition of any new or change to any existing UI. May I get your approval on Accessibility? Thanks!

**Comment by pgunapal** [comment]
@samwatibm Hi Sam, The newly added messages are translated and are in the master branch. May I get your approval for globalization?

**Comment by pgunapal** [comment]
@skasund Hi Swati, I have completed and uploaded the STE for mpHealth-3.0 (https://ibm.box.com/s/zot7dl6x1k53sp9kn0nz9njtuh3fxtuz). May I get your approval for STE?

**Comment by pgunapal** [comment]
@donbourne **Serviceability Approval Comment** - Please answer the following questions for serviceability approval:

1.  UFO -- does the UFO identify the most likely problems customers will see and identify how the feature will enable them to diagnose and solve those problems without resorting to raising a PMR?  Have these issues been addressed in the implementation?

- Yes, the UFO includes the most likely problems the customers will see, which is usually in their own implementation of the HealthCheck interface, in their applications. Detailed messages will be logged in the messages.log. Major functional changes  were removal of deprecated API and a method rename, its implementation is similar to mpHealth-2.x feature.

2.  Test and Demo -- As part of the serviceability process we're asking feature teams to test and analyze common problem paths for serviceability and demo those problem paths to someone not involved in the development of the feature (eg. L2, test team, or another development team).  
a) What problem paths were tested and demonstrated?
b) Who did you demo to?
c) Do the people you demo'd to agree that the serviceability of the demonstrated problem scenarios is sufficient to avoid PMRs for any problems customers are likely to encounter, or that L2 should be able to quickly address those problems without need to engage L3?

- For mpHealth-3.0, the major changes were a removal of the deprecated API (`@Health`) and rename of a method, the end-user scenario remain the same as mpHealth-2.x.

3.  SVT -- SVT team is often the first team to try new features and often encounters problems setting up and using them.  Note that we're not expecting SVT to do full serviceability testing -- just to sign-off on the serviceability of the problem paths they encountered.
a) Who conducted SVT tests for this feature?

- Brian Hanczaryk and team had done SVT testing for this feature, where they had updated their applications to use mpHealth-3.0 feature and there weren't any issues.

b) Do they agree that the serviceability of the problems they encountered is sufficient to avoid PMRs, or that L2 should be able to quickly address those problems without need to engage L3?  

- Yes.

4.  Which L2 / L3 queues will handle PMRs for this feature?  Ensure they are present in the contact reference file and in the queue contact summary, and that the respective L2/L3 teams know they are supporting it.  Ask Don Bourne if you need links or more info.
- WAS L3: Liberty Log Analytics

**Comment by skasund** [comment]
@pgunapal   Thank you for the STE slides.  I've approved the feature.

**Comment by chirp1** [comment]
ID issue is https://github.com/OpenLiberty/docs/issues/3292 Prashanth and I worked out and agreed on the description. Approving.
